src/main/java/org/jboss/arquillian/testcontainers/TestContainersObserver.java

```diff
@@ -49,21 +50,35 @@ public void stopContainer(@Observes AfterClass afterClass) {
 
     private void checkForDocker(boolean failIfNoDocker) {
         final String detailMessage = "No Docker environment is available.";
-        if (!isDockerAvailable() && failIfNoDocker) {
-            throw new AssertionError(detailMessage);
-        } else {
-            try {
-                Class<?> clazz = Class.forName("org.junit.AssumptionViolatedException");
-                Constructor<?> ctor = clazz.getDeclaredConstructor(String.class);
-                throw (RuntimeException) ctor.newInstance(detailMessage);
-            } catch (Exception e) {
-                // TestNG?
+        if (!isDockerAvailable()) {
+            if (failIfNoDocker) {
+                throw new AssertionError(detailMessage);
+            } else if (!isDockerAvailable()) {
```

Comment:
Why is this checked twice? We already know the answer here, right?

Reply:
Because I forgot to remove that :) Thank you!